### ISSUE TITLE
Unexpected behavior when binding multiple commands to one key

### ISSUE THREAD

**Issue by norstbox** [mpv-player/mpv#7841]
### Important Information

- mpv-x86_64-20200614 c7fe4ae
- Windows 8.1
- Windows builds by [shinchiro](https://sourceforge.net/projects/mpv-player-windows/files)

### Reproduction steps

1. Bind multiple commands to one key in input.conf. For exampe, `s cycle sub; cycle audio`
2. Press that key, wait one second and release it.

### Expected behavior
Each command is executed only once. For the above example, next subtitle and audio tracks will be selected

### Actual behavior
Each command is executed many times, as with `repeatable` prefix. For the above example, subtitle and audio tracks will be switched multiple times in cycle.

### Log file
[output.txt](https://github.com/mpv-player/mpv/files/4797915/output.txt)

### Sample files
Any media file

**Comment by norstbox** [comment]
Also I notice that `repeatable` prefix doesn't change behaviour of the keybinding for the single `cycle` command. I.e. property is changed only once, to the next value. Not sure is this a bug, but at least seems not consistent with the man:
> `repeatable` This prefix forces enabling key repeat in any case.

**Comment by Dudemanguy** [comment]
I think the usage of `;` makes mpv interpret this key as `repeatable` which is why you get the behavior you see (cycling while holding the key down). That may or may not be desirable depending on the person I suppose.

**Comment by eric-saintetienne** [comment]
`;` do separate commands es explained in the documentation at https://mpv.io/manual/master/#input-conf-syntax

**Comment by Dudemanguy** [comment]
That doesn't contradict what I said.

**Comment by eric-saintetienne** [comment]
You're still wrong. But if you don't think so, then please quote where it's stated in the documentation.

**Comment by Dudemanguy** [comment]
What I was originally pointing out is that the logic that implements executing multiple commands causes mpv internally to treat it as one repeatable command which is why OP gets the behavior he sees.

**Comment by eric-saintetienne** [comment]
Which is different from what you wrote in your message:

_> I think the usage of `;` makes mpv interpret this key as repeatable_

**Comment by avih** [comment]
Currently the mpv code indeed considerd any list of commands to be repeatable. This was added in 2014 to fix #807 .

I'll try to change it so that the repeatability of a list is controlled only by the first command.

**Comment by eric-saintetienne** [comment]
Let me as: Is that due to the semi colon like Dude Man Guy thinks?

**Comment by avih** [comment]
> Is that due to the semi colon like Dude Man Guy thinks?

Implicitly, yes. Technically, when mpv tries to determine if a command is repeatable, and that command is a list of commands (separated by semicolons, because there's no other way to specify a list of commands), then mpv answers "yes, it's repeatable". It's currently a hardcoded behavior.

As I said, I'll try to change it so that only the first command in a list determines the repeatability.

**Comment by eric-saintetienne** [comment]
In the meantime it maybe quickto update the documentation about semicolon with the actual behaviour.

**Comment by avih** [comment]
No point in that. Hopefully I'll make this change soon, including the docs to the new behavior.

**Comment by avih** [comment]
> Also I notice that `repeatable` prefix doesn't change behaviour of the keybinding for the single `cycle` command. I.e. property is changed only once, to the next value. Not sure is this a bug, but at least seems not consistent with the man:
> 
> > `repeatable` This prefix forces enabling key repeat in any case.

Can't reproduce. Tested:
```
x repeatable cycle-values volume 20 50 80
```
and holding `x` cycles repeatedly.

If the issue still exists for you, please file a new issue.

**Comment by avih** [comment]
> confirmed, tested with:
> 
> ```
> x repeatable cycle osd-level
> ```
> 
> and holding `x` cycles once, and doesn't repeat.

I've looked at the code, and it seems... complex with some guesswork.

On one hand, the cycle command has `.allow_auto_repeat == true`, and OTOH the command handler itself disables auto repeat depending on the property which is being cycled, according to some hueristics.

However, the new code which makes lists repeatable if the first command is repeatable can work around it, like so:
```
x repeatable ignore; cycle osd-level
```

So even if you open an issue, I don't know if anyone will find the time and energy to address it. I don't think the solution is simple here.

But at least the workaround seems to work.
